scripts/run_backend_tests_test.py

```diff
@@ -0,0 +1,888 @@
+
+    def test_failed_test_suite_throws_error(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+
+        task1 = MockTask()
+        task1.exception = Exception(
+            'Test suite failed: 6 tests run, 0 errors, '
+            '2 failures')
+
+        tasks = [task1]
+        task_to_taskspec = {}
+        test_target = 'scripts.new_script.py'
+        task_to_taskspec[tasks[0]] = run_backend_tests.TestingTaskSpec( # type: ignore[no-untyped-call]
+            test_target, False)
+
+        with self.print_swap:
+            run_backend_tests._check_test_results( # type: ignore[no-untyped-call] # pylint: disable=protected-access
+                tasks, task_to_taskspec, False)
+
+        self.assertIn(
+            'FAILED    %s: %s errors, %s failures' % (test_target, 0, 2),
+            self.print_arr)
+
+    def test_tests_failed_due_to_internal_error(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+
+        task1 = MockTask()
+        task1.exception = Exception('Some internal error.')
+
+        tasks = [task1]
+        task_to_taskspec = {}
+        test_target = 'scripts.new_script.py'
+        task_to_taskspec[tasks[0]] = run_backend_tests.TestingTaskSpec( # type: ignore[no-untyped-call]
+            test_target, False)
+
+        with self.print_swap, self.assertRaisesRegex(
+            Exception, 'Some internal error.'
+        ):
+            run_backend_tests._check_test_results( # type: ignore[no-untyped-call] # pylint: disable=protected-access
+                tasks, task_to_taskspec, False)
+
+        self.assertIn(
+            '    WARNING: FAILED TO RUN %s' % test_target, self.print_arr)
+        self.assertIn(
+            '    This is most likely due to an import error.', self.print_arr)
+
+    def test_unfinished_tests_are_cancelled(self) -> None:
+        with self.swap_install_third_party_libs:
+            from scripts import run_backend_tests
+
+        task = MockTask()
+        task.finished = False
+        task_output = ['Ran 9 tests in 1.244s', '98']
+        task_result = concurrent_task_utils.TaskResult( # type: ignore[no-untyped-call]
+            'task1', False, task_output, task_output)
+        task.task_results.append(task_result)
+
+        tasks = [task]
+        task_to_taskspec = {}
+        test_target = 'scripts.new_script.py'
+        task_to_taskspec[tasks[0]] = run_backend_tests.TestingTaskSpec( # type: ignore[no-untyped-call]
+            test_target, False)
+
+        with self.print_swap:
+            run_backend_tests._check_test_results( # type: ignore[no-untyped-call] # pylint: disable=protected-access
+                tasks, task_to_taskspec, True)
+
+        self.assertIn('CANCELED  %s' % test_target, self.print_arr)
+
+    def test_incomplete_coverage_is_calculated_correctly(self) -> None:
```

Comment:
This isn't really checking the calculation of coverage, is it? The coverage has already been calculated and is being provided as the string `'98'`

Reply:
Yes. Changed the method name now.